app/models/requirement_template.rb

```diff
@@ -0,0 +1,14 @@
+class RequirementTemplate < ApplicationRecord
+  belongs_to :activity, class_name: "Activity"
+  belongs_to :permit_type, class_name: "PermitType"
+
+  validates :activity_id, uniqueness: { scope: :permit_type_id }
+  validate :ensure_correct_type_for_permit_type_and_activity
+
+  private
+
+  def ensure_correct_type_for_permit_type_and_activity
+    errors.add(:permit_type, "must be of type PermitType") unless permit_type.type == "PermitType"
+    errors.add(:activity, "must be of type Activity") unless activity.type == "Activity"
+  end
```

Comment:
@bkmaibach does it work without these validations? Asking because we have something similar in productmotion where TreeItem is the base class and Feature and Collection inherit from it. It seems like rails is able to automatically handle this in pm code base, so wondering if the same can be achieved here

Reply:
probably